yage-gl/src/objects/buffer.rs

```diff
@@ -0,0 +1,66 @@
+use crate::{GL, GlFunctions};
+
+/// Wrapper around an OpenGL array or element array buffer. 
+pub struct Buffer<'a> {
+    gl: &'a GL,
+    /// Target for use in `glBindBuffer`
+    target: u32,
+    vbo: <GL as GlFunctions>::GlBuffer,
+}
+
+impl<'a> Buffer<'a> {
+    /// Creates an empty buffer. 
+    /// 
+    /// # Parameters
+    /// - `gl`: GL context
+    /// - `target`: must be a valid glenum for `glBindBuffer`
+    pub fn new(gl: &'a GL, target: u32) -> Self {
+        Self {
+            gl,
+            target,
+            vbo: gl.create_buffer()
+        }
+    }
+
+    /// Creates the buffer object's data store.
+    /// 
+    /// # Parameters
+    /// - `data`: buffer data
+    /// - `usage`: must be a valid glenum for `glBufferData`
+    pub fn set_data<T>(&self, data: &[T], usage: u32) {
+        self.gl.buffer_data(self.target, data, usage);
```

Comment:
Does this not require a bind first? Otherwise calling this on a buffer without binding it first would not have the intended effect

Reply:
Yes. As far as I saw, globjects doesn't bind automatically in this case; and webgl-operate has optional boolean params everywhere for bind, unbind. Since Rust doesn't have optional arguments, I didn't want to add those and leave it up to the user (as far as I understand, skipping unnecessary binding/unbinding is a good idea...).
If we leave like this, I'd extend the doc string to mention this.

Reply:
Ah, I see `BufferImplementation_Legacy.cpp` does bind automatically...
I'm not sure - I guess in this case unnecessary binds are unlikely so `set_data` could do it. But in other cases it definitely should not, e.g. https://github.com/cginternals/webgl-examples/blob/be7f57bef33ca85bb60c97fe81793da307a0482e/source/gltf/material.ts#L135-L168.

Reply:
I think we should bind the buffer automatically, otherwise using these objects would be very awkward. And I think this should be the default in most cases. Lets talk about the cases where this is not wanted when we get to them.

Reply:
Ok - the first case is `attrib_enable` a few lines down which I also want to implement very soon. The triangle example will call both `set_data` and `attrib_enable` when updated and I'd really like to avoid a repeated bind, because in WebAssembly each GL call is 'expensive' (..calling into JS).

Reply:
I'd propose having a variable `is_bound` on the struct that tracks the state.

Reply:
Yes, but when do we reset is_bound then (without global state etc.)?

Reply:
Ah, right, didn't think till the end...
In that case I'm not quite convinced automatic binding is a good idea.